src/Reader.ts

```diff
@@ -0,0 +1,485 @@
+    return this.navigator.mostRecentNavigatedTocItem();
+  };
+  totalResources = () => {
+    return this.navigator.totalResources();
+  };
+
+  /**
+   * Settings
+   */
+  get publicationLanguage() {
+    return this.navigator.publication.metadata.language;
+  }
+  resetUserSettings = async () => {
+    return await this.settings.resetUserSettings();
+  };
+  applyUserSettings = async (userSettings) => {
+    return await this.settings.applyUserSettings(userSettings);
+  };
+  currentSettings = () => {
+    return this.settings.currentSettings();
+  };
+  scroll = async (value) => {
+    return await this.settings.scroll(value);
+  };
+
+  /**
+   * pitch?
+   */
+  increase = (incremental) => {
+    if (
+      (incremental === "pitch" ||
+        incremental === "rate" ||
+        incremental === "volume") &&
+      this.navigator.rights?.enableTTS
+    ) {
+      this.ttsSettings.increase(incremental);
+    } else {
+      this.settings.increase(incremental);
+    }
+  };
+  decrease = (incremental) => {
+    if (
+      (incremental === "pitch" ||
+        incremental === "rate" ||
+        incremental === "volume") &&
+      this.navigator.rights?.enableTTS
+    ) {
+      this.ttsSettings.decrease(incremental);
+    } else {
+      this.settings.decrease(incremental);
+    }
+  };
+
+  /**
+   * Publisher?
+   */
+  publisher = (on) => {
+    this.settings.publisher(on);
+  };
+
+  /**
+   * TTS Settings
+   */
+  resetTTSSettings = () => {
+    if (this.navigator.rights?.enableTTS) {
+      this.ttsSettings.resetTTSSettings();
+    }
+  };
+  applyTTSSettings = (ttsSettings) => {
+    if (this.navigator.rights?.enableTTS) {
+      this.ttsSettings.applyTTSSettings(ttsSettings);
+    }
+  };
+
+  applyTTSSetting = (key, value) => {
+    if (this.navigator.rights?.enableTTS) {
+      this.ttsSettings.applyTTSSetting(key, value);
+    }
+  };
+  applyPreferredVoice = (value) => {
+    if (this.navigator.rights?.enableTTS) {
+      this.ttsSettings.applyPreferredVoice(value);
+    }
+  };
+
+  /**
+   * Navigation
+   * @TODO : These should return promises that complete when they are done.
+   */
+  get currentLocator() {
+    return this.navigator.currentLocator();
+  }
+  get positions() {
+    return this.navigator.positions();
+  }
+  goTo = async (locator) => {
+    this.navigator.goTo(locator);
+  };
+  goToPosition = async (value) => {
+    return this.navigator.goToPosition(value);
+  };
+  nextResource = async () => {
+    this.navigator.nextResource();
+  };
+  previousResource = async () => {
+    this.navigator.previousResource();
+  };
+  nextPage = async () => {
+    this.navigator.nextPage();
+  };
+  previousPage = async () => {
+    this.navigator.previousPage();
+  };
+  atStart = async () => {
+    return this.navigator.atStart();
+  };
+  atEnd = async () => {
+    return this.navigator.atEnd();
+  };
+  snapToElement = async (value) => {
+    this.navigator.snapToElement(value);
+  };
+
+  /**
+   * ??
```

Comment:
You have `attributes` in the reader when you initialize it. You can set `margin`, `navigationHeight` etc... 

This is in case you change the attributes after initializing the reader.

`IframeAttribute`